RecoBTag/PixelCluster/plugins/PixelClusterTagInfoProducer.cc

```diff
@@ -0,0 +1,262 @@
+//
+/**\class PixelClusterTagInfoProducer PixelCluster RecoBTag/PixelCluster/plugins/PixelClusterTagInfoProducer.cc
+
+ Description: Produces a collection of PixelClusterTagInfo objects,
+  that contain the pixel cluster hit multiplicity in each pixel layer or disk
+  in a narrow cone around the jet axis.
+
+ Implementation:
+     If the event does not fulfill minimum conditions (at least one jet above threshold,
+     and a valid primary vertex) and empty collection is filled. Otherwise, a loop over
+     the pixel cluster collection fills a vector of reco::PixelClusterProperties that
+     contains the geometrical position and the charge of the cluster above threshold.
+     A second loop on jets performs the dR association, and fills the TagInfo collection.
+*/
+//
+// Original Author:  Alberto Zucchetta, Manuel Sommerhalder (UniZ) [zucchett]
+//         Created:  Wed, 03 Jul 2019 12:37:30 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+// TagInfo
+#include "DataFormats/BTauReco/interface/PixelClusterTagInfo.h"
+
+// For vertices
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+
+// For jet
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "DataFormats/JetReco/interface/PFJetCollection.h"
+
+// For pixel clusters
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+
+// Pixel topology
+#include "DataFormats/SiPixelDetId/interface/PixelSubdetector.h"
+#include "DataFormats/SiPixelDetId/interface/PixelBarrelName.h"
+#include "DataFormats/SiPixelDetId/interface/PixelEndcapName.h"
+#include "DataFormats/SiPixelDetId/interface/PXBDetId.h"
+#include "DataFormats/SiPixelDetId/interface/PXFDetId.h"
+
+// Geometry
+#include "Geometry/Records/interface/IdealGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetUnit.h"
+#include "Geometry/TrackerGeometryBuilder/interface/PixelGeomDetType.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/CommonDetUnit/interface/GeomDetType.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+
+// ROOT
+#include "TVector3.h"
+#include "TLorentzVector.h"
```

Comment:
please switch to non-TObject vector variants.
See e.g.  in DataFormats/Math/interface

Reply:
Ok, we replaced both TVector3 with GlobalPoints and reco::deltaR function.